.travis.yml

```diff
@@ -6,20 +6,19 @@ install:
   - nvm install node
   - npm install -g source-map-support
   - go get -u github.com/gopherjs/gopherjs
-  - go get -u github.com/golang/lint/golint
+  - go get -u gopkg.in/alecthomas/gometalinter.v1
```

Comment:
I've just realized a second benefit to this approach. Imagine a situation where a user has some very complex markup for many different logical cases (e.g. themes or something like that) and they want to split it out into separate functions:

```Go
func GetTheme(name string) []Applyer {
	...
}
```

Callers would have to use it like this:

```Go
return elem.Div(
	vecty.Markup(foo.GetTheme("dark")...),
	...
)
```

Which, if you're a complete beginner, does look a bit strange. This is because `GetTheme` can't return the actual value of `vecty.Markup` to the user of the package. In contrast, with it being exported as suggested above, they could write:

```Go
func GetTheme(name string) vecty.MarkupList {
	return vecty.Markup(getTheme("dark")...)
}
```

And then callers may just use it like:

```Go
return elem.Div(
	foo.GetTheme("dark"),
	...
)
```

Reply:
Just to clarify a few things about `gometalinter`.  It's been around for a long time, and is pretty widely used.

1. It _vendors_ checks so that versions are stable, and installs always succeed, but they've had plenty of changes upstreamed to those projects.  megacheck is able to get away with not vendoring since all the checks are written by the same author.
2. Honestly, I'm not sure whether new linters will be added to v1 or not, but being explicit about what checks are run seems fine to me either way, otherwise new checks may unexpectedly break your CI.
3. The config file is optional, I just used it to make the CI config clearer, since we special-case `golint` for the generated code.  Choosing which checks to run is already something every project does, whether the checks are bundled under a single runner or not, by choosing what to enable in the CI run - the box is already open ;-)
4. Not exactly, it just makes it easy to run multiple existing checks via a single executable call - this is particularly useful for editor integration.

megacheck is very new by comparison (though I'm sure the quality is high), wrapping some of Dominik's other tools, though they've been available via gometalinter for a long time.  I'm not really phased either way, but I thought some of the criticisms were a little hard on gometalinter.

---

As for the API suggestion here, I like it :+1: